Microsoft.Azure.Cosmos/src/Serializer/CosmosFeedResponseSerializer.cs

```diff
@@ -0,0 +1,139 @@
+        /// </summary>
+        /// <param name="serializerCore">The cosmos serializer</param>
+        /// <param name="streamWithServiceEnvelope">A stream with the service envelope like: { "ContainerRid":"Test", "Documents":[{ "id":"MyItem"}], "count":1}</param>
+        /// <param name="resourceType">The resource type</param>
+        /// <returns>A read only list of the serialized items</returns>
+        internal static IReadOnlyList<T> FromFeedResponseStream<T>(
+            CosmosSerializerCore serializerCore,
+            Stream streamWithServiceEnvelope,
+            Documents.ResourceType resourceType)
+        {
+            if (streamWithServiceEnvelope == null)
+            {
+                return new List<T>();
+            }
+
+            using (MemoryStream stream = GetStreamWithoutServiceEnvelope(
+                streamWithServiceEnvelope,
+                resourceType))
+            {
+                return serializerCore.FromStream<List<T>>(stream);
+            }
+        }
+
+        /// <summary>
+        /// The service returns feed responses in an envelope. This removes the envelope
+        /// so it only returns the array of items.
+        /// </summary>
+        /// <param name="streamWithServiceEnvelope">A stream with the service envelope like: { "ContainerRid":"Test", "Documents":[{ "id":"MyItem"}], "count":1}</param>
+        /// <param name="resourceType">The resource type</param>
+        /// <returns>A stream containing only the array of items</returns>
+        internal static MemoryStream GetStreamWithoutServiceEnvelope(
+            Stream streamWithServiceEnvelope,
+            Documents.ResourceType resourceType)
+        {
+            ReadOnlyMemory<byte> content;
+            MemoryStream? memoryStreamWithEnvelope = streamWithServiceEnvelope as MemoryStream;
+            if (memoryStreamWithEnvelope == null)
+            {
+                memoryStreamWithEnvelope = new MemoryStream();
+                streamWithServiceEnvelope.CopyTo(memoryStreamWithEnvelope);
+                memoryStreamWithEnvelope.Position = 0;
+            }
+
+            if (memoryStreamWithEnvelope.TryGetBuffer(out ArraySegment<byte> buffer))
+            {
+                content = buffer;
+            }
+            else
+            {
+                content = memoryStreamWithEnvelope.ToArray();
+            }
+
+            int start = GetArrayStartPosition(content, resourceType);
+            int end = GetArrayEndPosition(content, resourceType);
+
+            ReadOnlyMemory<byte> spanwithOnlyArray = content.Slice(start, end - start + 1);
+            if (!MemoryMarshal.TryGetArray(spanwithOnlyArray, out ArraySegment<byte> resultAsArray))
+            {
+                resultAsArray = new ArraySegment<byte>(spanwithOnlyArray.ToArray());
+            }
+
+            MemoryStream arrayOnlyStream = new MemoryStream(resultAsArray.Array, resultAsArray.Offset, resultAsArray.Count);
```

Comment:
This might not have any impact, it really depends how the serializer is internally accessing the stream, but might be worth it:
```suggestion
            MemoryStream arrayOnlyStream = new MemoryStream(resultAsArray.Array, resultAsArray.Offset, resultAsArray.Count, writable: false, publiclyVisible: true);
```